pandas/_libs/lib.pyx

```diff
@@ -538,6 +538,24 @@ def has_infs(floating[:] arr) -> bool:
     return ret
 
 
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def has_NA(ndarray arr) -> bool:
```

Comment:
ndarray[object]?

Reply:
When I try that (or `object[:]`), end up getting errors like `ValueError: Buffer dtype mismatch, expected 'Python object' but got 'double'`

Reply:
yah, this should only be called on object-dtype, in other cases we already know the result is False

Reply:
Ah got it thanks, have changed